### PR TITLE
Fix update button and certificate type toggle issues in SAML web application

### PR DESCRIPTION
**Pull request by shashimalcse**
### Purpose
Update button is not working when toggle between provide certificate and none certificate types in SAML web application.  Also user can toggle back to none type in the certification section if request signature validation is enabled. 

These issues are fixed in this PR.

### Related Issues
- Issue `#1` or (None)

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: modules/forms/src/components/field.tsx [hunk 1]
```diff
@@ -244,6 +244,9 @@ export const InnerField = React.forwardRef((props: InnerFieldPropsInterface, ref
                                     value={ value }
                                     checked={ form.get(inputField.name) === value }
                                     onChange={ (event: React.ChangeEvent<HTMLInputElement>, { value }) => {
+                                        if(inputField?.onBefore && !inputField.onBefore(event,value)){
```
Comment (yasinmiran) [inline modules/forms/src/components/field.tsx:247]: Use ESLint for formatting

```suggestion
                                        if (inputField?.onBefore && !inputField.onBefore(event, value)) {
```

File: apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx [hunk 2]
```diff
@@ -395,12 +440,17 @@ export const ApplicationCertificateWrapper: FunctionComponent<ApplicationWrapper
...
                 </Forms>
             )
             : null
     );
 };
 
+
+
+
+
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx:453]: Remove these empty lines.

```suggestion
```

File: apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx [hunk 3]
```diff
@@ -255,6 +287,19 @@ export const ApplicationCertificateWrapper: FunctionComponent<ApplicationWrapper
                                         updateCertType(values.get("certificateType") as CertificateTypeInterface);
                                     }
                                 }
+                                onBefore={
+                                    (event,value) => {
+                                        const Certtype  =  value as CertificateTypeInterface;
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx:292]: Use camel case.

```suggestion
                                        const certType  =  value as CertificateTypeInterface;
```

File: apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx [hunk 4]
```diff
@@ -77,6 +77,7 @@ interface ApplicationWrapperCertificatesPropsInterface extends TestableComponent
     hidden: boolean;
     readOnly: boolean;
     triggerSubmit?: boolean;
+    canDiscardCerticate?: () => boolean;
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx:80]: Correct the spellings of this variable

```suggestion
    canDiscardCertificate?: () => boolean;
```

File: apps/console/src/features/applications/components/forms/inbound-saml-form.tsx [hunk 5]
```diff
@@ -1775,6 +1777,7 @@ export const InboundSAMLForm: FunctionComponent<InboundSAMLFormPropsInterface> =
                                     application={ application }
                                     updateCertFinalValue={ setFinalCertValue }
                                     updateCertType={ setSelectedCertType }
+                                    canDiscardCerticate = { ():boolean => !isRequestSignatureValidationEnabled }
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/forms/inbound-saml-form.tsx:1780]:
```suggestion
                                    canDiscardCerticate = { (): boolean => !isRequestSignatureValidationEnabled }
```

File: apps/console/src/features/applications/components/forms/inbound-saml-form.tsx [hunk 6]
```diff
@@ -476,6 +475,9 @@ export const InboundSAMLForm: FunctionComponent<InboundSAMLFormPropsInterface> =
                             setAssertionConsumerUrlError(true);
                             scrollToInValidField("consumerURL");
                         } else {
+                            if(selectedCertType === CertificateTypeInterface.NONE){
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/forms/inbound-saml-form.tsx:478]:
```suggestion
                            if (selectedCertType === CertificateTypeInterface.NONE) {
```

**Review by yasinmiran — CHANGES_REQUESTED** [review]

File: apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx [hunk 7]
```diff
@@ -255,6 +287,19 @@ export const ApplicationCertificateWrapper: FunctionComponent<ApplicationWrapper
                                         updateCertType(values.get("certificateType") as CertificateTypeInterface);
                                     }
                                 }
+                                onBefore={
+                                    (event,value) => {
+                                        const certType  =  value as CertificateTypeInterface;
```
Comment (yasinmiran) [inline apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx:292]: Use ESLint auto-formatting.

```suggestion
                                        const certType  =  value as CertificateTypeInterface;
```

**Review by yasinmiran — APPROVED** [review]

**Comment by senthalan** [comment]
Closing as the PR is in the open state for a while. Please feel free to reopen and merge if needed.

File: apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx [hunk 8]
```diff
@@ -255,6 +287,19 @@ export const ApplicationCertificateWrapper: FunctionComponent<ApplicationWrapper
...
+                                onBefore={
+                                    (event,value) => {
+                                        const certType = value as CertificateTypeInterface;
+
+                                        if(CertificateTypeInterface.NONE === certType && !canDiscardCertificate()){
+                                            setShowInvalidOperationModal(true);
+                                            
+                                            return false;
+                                        }
+
```
Comment (lashinijay) [inline apps/console/src/features/applications/components/settings/certificate/application-certificate-wrapper.tsx:299]: Shall we remove these unwanted new lines?

File: apps/console/src/features/applications/components/forms/inbound-saml-form.tsx [hunk 9]
```diff
@@ -1775,6 +1777,7 @@ export const InboundSAMLForm: FunctionComponent<InboundSAMLFormPropsInterface> =
                                     application={ application }
                                     updateCertFinalValue={ setFinalCertValue }
                                     updateCertType={ setSelectedCertType }
+                                    canDiscardCertificate = { (): boolean => !isRequestSignatureValidationEnabled }
```
[the remark this answers is not in the document]
Comment (shashimalcse) [reply apps/console/src/features/applications/components/forms/inbound-saml-form.tsx:1780]: If there is a possibility that the conditions for discarding a certificate might become more complex in the future, having a function allows you to easily modify the logic right?

**Pull request merged**